server/order/controller/product_controller.js

```diff
@@ -5,8 +5,17 @@
 const Product = require('../model/product');
 
 exports.addProduct = function (productData) {
-  const product = new Product(JSON.parse(productData));
-  product.save();
+  return new Promise((resolve, reject) => {
+    Product.findOneAndUpdate({ name: productData.name }
```

Comment:
Is the name a unique key? If not so, this logic is incorrect.
Also, this is an add function. Then, do we really need to find something?